src/engine/threaded_engine_perdevice.cc

```diff
@@ -87,21 +92,34 @@ class ThreadedEnginePerDevice : public ThreadedEngine {
         int nthread = gpu_worker_nthreads_;
         int dev_id = ctx.dev_id;
         if (is_copy) {
+          auto ptr =
```

Comment:
how often? I tested and only see it getting called about 4 times before the training iterations of mnist.